### ISSUE TITLE
packages/sdk: update to support project key selection other than Url

### ISSUE THREAD

**Issue by adnasa** [commercetools/merchant-center-application-kit#1773]
**Is your feature request related to a problem? Please describe.**

Yes.
I attempt to use `msw` to mock my requests as I run integration tests (using RTL) on a view that renders `Sdk.Get`.
When doing so, I get the following error: 

```
Expected projectKey to be defined for action service "states"
```

reference: https://github.com/commercetools/merchant-center-application-kit/blob/master/packages/sdk/src/middleware/middleware.ts#L32:L32

This is because in the background, my view runs an sdk action `sdk.get`.
The `sdk.get` is picked up by the redux middleware and attempts to build a URI from my sdk action.
The middleware requires a projectKey to successfully build a URI, with the help of a `getProjectKey` util. The default definition of `getProjectKey` is `selectProjectKeyFromUrl` which uses `window.location.pathname`. This value is undefined.

**Describe the solution you'd like**

I would like to avoid having to mock window, for my request to succeed. This is additional boilerplate to my application test that I don't have to set up.

What would be preferable is if I can pass a `projectKey` (that for example can derive from `match.params.projectKey` given that I use `react-router`) or that I pass a `getProjectKey` that overrides `selectProjectKeyFromUrl`.

**Describe alternatives you've considered**

update `test-util` to use `react-router` context to pick the projectKey value.
Maybe it can be done here
https://github.com/commercetools/merchant-center-application-kit/blob/master/packages/application-shell/src/test-utils/test-utils.tsx#L495:L495

---

**Additional Context**:

- Original discussion in this PR: #1772 
- This blocks removing use of `enzyme` for the `Sdk.Get` component

**Comment by adnasa** [comment]
**More context**

The original problem was discovered when I tried to migrate [this test](https://github.com/commercetools/merchant-center-application-kit/blob/master/playground/src/components/entry-point/entry-point.spec.js#L37:L37) into using `msw`.

I tried this approach where I forward `getProjectKey` from `renderAppWithRedux -> configure-store -> createSdkMiddleware` (falls back to `selectProjectKeyFromUrl`) and the test could proceed.

example 

```js
const render = (options) => {
  return renderAppWithRedux(<ApplicationStateMachines />, {
    permissions: {
      canViewDeveloperSettings: true,
      canManageDeveloperSettings: true,
    },
    // to forward to the redux store.
    getProjectKey: () => projectKey,
    ...options,
  });
};
```

the `test-util` could fallback to `defaultProject.key` if a `getProjectKey` wasn't provided.
the middleware could pick up the provided `projectKey`, however the request wasn't issued with or without `msw` mocking requests.

here is my attempt: https://gist.github.com/adnasa/ece87d48288d2afffb00d4e6d8a906fb

oh well, experiment 1 ❌
